common/autocomplete/awesomplete.js

```diff
@@ -0,0 +1,205 @@
+     */
+    _selectionHandler(value) {
+        const {selectionHandler} = this.props;
+        if (selectionHandler) {
+            const {pickList} = this.props;
+            const selectedPick = find(pickList, {value});
+            selectionHandler(selectedPick);
+        }
+        this._isSelecting = true; // Private flag to tell the blur listener not to replace the value
+        this.setState({value});
+    },
+    /**
+     * Extract list of suggestions from pick list
+     * @param  {Object} data the pick list
+     * @return {Array}      the suggestion array
+     */
+    _extractListFromData(data) {
+        return data.map(datum => datum.value);
+    },
+    /**
+     * Get code from value in the pick list
+     * @param  {String} value the value
+     * @return {String} the code
+     */
+    _getCodeFromValue(value) {
+        const {pickList} = this.props;
+        const pick = find(pickList, {value});
+        return pick ? pick.code : pick;
+    },
+    /**
+     * Get value from code in the pick list
+     * @param  {String} code the code
+     * @param  {Object} pickList=this.props.pickList  optional pick list to resolve the value from
+     * @return {String} value
+     */
+    _getValueFromCode(code, pickList=this.props.pickList) {
+        const pick = find(pickList, {code});
+        return pick ? pick.value : '';
+    },
+    /**
+     * Get the current code
+     * @return {String} the code
+     */
+    getValue() {
+        const {value} = this.state;
+        return this._getCodeFromValue(value);
+    },
+    /**
+     * On input blur.
+     * If validate is set in the props, validate the current value and erase it if not valid.
+     */
+    _onInputBlur() {
+        const {value} = this.state;
+        const {validate} = this.props;
+        const code = this._getCodeFromValue(value);
+        if (!code && validate && !this._isSelecting) {
+            this.setState({value: ''});
+        }
+        this._isSelecting = false;
+    },
+    /**
+     * On input change
+     * @param  {Object} event change event
+     */
+    _onInputChange(event) {
+        const {value} = event.target;
+        this.setState({value});
+        if (this._changeTimeout) {
+            clearTimeout(this._changeTimeout);
+        }
+        this._changeTimeout = setTimeout(() => {
+            let {inputChangeHandler} = this.props;
+            if (inputChangeHandler) {
+                inputChangeHandler(value);
+            }
+        }, 200);
+    },
+    /**
+     * Render the edit mode
+     * @return {HTML} rendered element
+     */
+    _renderEdit() {
+        let {value} = this.state;
+        let {_onInputBlur, _onInputChange} = this;
+        return (
+            <input onBlur={_onInputBlur} onChange={_onInputChange} ref='input' value={value}/>
+        );
+    },
+    /**
+     * Render the consult mode
+     * @return {HTML} rendered element
+     */
+    _renderConsult() {
```

Comment:
See if there is two separate components, for now it seems right to let the component render the consult itself.